File: week-02/day-02/matrix.py

# - Create (dynamically) a two dimensional list
#   with the following matrix. Use a loop!
#
#   1 0 0 0
#   0 1 0 0
#   0 0 1 0
#   0 0 0 1
#
# - Print this two dimensional list to the output

def matrix2d(x, y):
    matrix = []
    for i in range(x):
        row = []
        for j in range(y):
            if i == j:
                row.append(1)
            else:
                row.append(0)
        matrix.append (row)
    return matrix

for row in matrix2d(4, 4):
    for elem in row:
        print(elem, end=' ')     
    print()

File: week-07/day-02/matrix.js

'use strict';
// - Create (dynamically*) a two dimensional list
//   with the following matrix**. Use a loop!
//
//   0 0 0 1
//   0 0 1 0
//   0 1 0 0
//   1 0 0 0
//
// - Print this two dimensional list to the console
//
// * size should depend on a variable
// ** Relax, a matrix is just like an array

function matrix2d(x){
    let matrix = [];
    for (let i = 0; i < x;  i++){
        let row = [];
        for (let j = x - 1; j >= 0; j--){
            if (i === j){
                row.push('1');
            }else{
                row.push('0');
            }
        }
        row = row.join(' ');
        matrix.push(row);
    }
    matrix = matrix.join('\n');
    console.log(matrix);
}

matrix2d(4);